async-disposable.js

```diff
@@ -379,73 +482,20 @@ export const AsyncDisposable = /** @type {DisposableConstructor} */ (
       };
     }
 
-    static [Symbol.asyncIterator]() {
-      /** @type {DisposableAggregate | undefined} */
-      let res = new (this || Disposable)();
```

Comment:
Not 100% refactor I guess, I fixed a previous copy-pasta mistake.

`Disposable` -> `AsyncDisposable`